File: src/main/java/uk/ac/napier/sa/controller/RemoteController.java
package uk.ac.napier.sa.controller;

import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;
import uk.ac.napier.sa.model.RemoteDatabaseManager;

public interface RemoteController {
    String retrieveProduct(int id);

    @NotNull
    String changePrice(int id, double newPrice);

    @NotNull
    String purchase(int customer, int product);

    String enrolOnLoyaltyCardScheme(int id);

    @Contract(pure = true)
    @NotNull
    String finance(char choice);

    void printLastNPurchases(int n);

    String generateReport();

    RemoteDatabaseManager rdbm();

    String stockMonitor();

    String addSale(int id, int selected);
}

File: src/main/java/uk/ac/napier/sa/controller/RemoteController.kt
package uk.ac.napier.sa.controller

import org.jetbrains.annotations.Contract
import uk.ac.napier.sa.model.RemoteDatabaseManager

interface RemoteController {
    fun retrieveProduct(id: Int): String?
    fun changePrice(id: Int, newPrice: Double): String
    fun purchase(customer: Int, product: Int): String
    fun enrolOnLoyaltyCardScheme(id: Int): String?

    @Contract(pure = true)
    fun finance(choice: Char): String
    fun printLastNPurchases(n: Int)
    fun generateReport(): String?
    fun rdbm(): RemoteDatabaseManager?
    fun stockMonitor(): String?
    fun addSale(id: Int, selected: Int): String?
}